v1/ast/compile.go

```diff
@@ -1808,6 +1804,20 @@ func (c *Compiler) err(errs ...*Error) bool { // returns if we should continue
 	return !isLimitReachedInThisCall // Return false if the limit was reached, true otherwise.
 }
 
+func (c *Compiler) forEachModule(f func(mod *Module)) {
+	if len(c.sorted) == 1 {
+		for name := range c.Modules {
+			f(c.Modules[name])
+			return
+		}
+	}
+	g := &errgroup.Group{}
+	for _, name := range c.sorted {
```

Comment:
Consider limiting the errgroup concurrency when processing large slices to avoid excessive goroutine creation. You could use `g.SetLimit(runtime.GOMAXPROCS(0))` or conditionally set a limit based on the slice size:
```
if len(c.sorted) > runtime.GOMAXPROCS(0) {
    g.SetLimit(runtime.GOMAXPROCS(0))
}
```
This will prevent spawning thousands of goroutines for large datasets and improve memory efficiency.

Reply:
Actually had considered that, but it didn't move the needle at all in my benchmarks. So I went back to the simple approach.